### DISCUSSION TITLE
Need original classes from Option

### DISCUSSION THREAD

**Discussion by TheoLyn69** [orchidjs/tom-select#40] [Q&A] [answered]
Hi,

I'm trying to integrate Tom-select library into my symfony project.
When my <SELECT> (and <OPTION>) elements are generated, I include some custom classes into my <OPTION> element (color and margin values). When I load the tom-select through my <SELECT> element, all my classes disappear. Only are keeped the tom-select ones.

How can I do to keep my classes ? I know there is a render function mentioned into the documentation : but it's seems to only work  with options that are created into the javascript ("options" member with list), not the original HTML <OPTION> elements...

Sorry, my english is not very good. Nevertheless, I hope you understand what I mean.

Thank you for your help.
Regards,

--
Sylvain

**Comment by oyejorge** [comment]
Hi Sylvain,

You could use a [custom render method](https://jsfiddle.net/3csv875b/) to add a custom class to all options

**Reply by TheoLyn69** [reply]
I know this functionality.

But, here is my generated code :

```
<Select>
<Option value="1" class="class1 class2">option 1</option>
<Option value="2" class="class1 class3">option 2</option>
<Option value="3" class="class4">option 3</option>
</Select>
```

How to keep the "class1", "class2" added on the previous <option> element...etc.... on the tom-select generated div item ??? With the render method, is it possible to access to the class list of the <option> element ?
Your reply is correct only if the class I add are constant...

Regards,
--
Sylvain

**Reply by oyejorge** [reply]
Hmm, I'm not sure this is possible right now without keeping track of class-option associations outside of tom-select. I have a change or two in mind that will make this much easier though.

**Reply by TheoLyn69** [reply]
Ok. Thanks for your reply.
At the moment, without this functionality, I can not use this library. In the future, I will keep an eye on tom-select library 👍 

Have a nice day,
Regards,
Sylvain

**Comment by oyejorge** [comment, marked as answer]
Alright, this can be done now with v1.4 like so:

```js
new TomSelect("#select-beast",{
	render: {
		option: function(data, escape) {
			return '<div class="' +escape(data.$option.className) +'">' + escape(data.text) + '</div>';
		},
	}
});
```

**Reply by shornikov** [reply]
ok, i update classes in select and run refreshOptions(). classes don't update (
